src/controllers/recipe.controller.ts

```diff
@@ -59,8 +57,10 @@ export class RecipeController {
 
   public async updateRecipe(req: Request, res: Response): Promise<Response> {
     try {
-      const { title, description, preparing, ingredients, url, id } =
-        matchedData(req);
+      const { title, description, preparing, ingredients } = matchedData(req, {
+        locations: ["body"],
+      });
+      const { id } = matchedData(req, { locations: ["params"] });
```

Comment:
Do you need to separate it like this? `matchedData(req)` should merge and return data from all locations. Perhaps you have duplicated `id` somewhere? Just a question, you don't have to change it.